src/IECore/FrameRange.cpp

```diff
@@ -58,6 +58,10 @@ FrameRange::FrameRange( Frame start, Frame end, Frame step ) : m_start( start ),
 	{
 		throw Exception( "FrameRange step cannot be zero" );
 	}
+	if ( step < 0 )
+	{
+		throw Exception( "FrameRange step cannot be negative. Consider using the reverse suffix instead." );
+	}
```

Comment:
Rejecting negative steps (rather than updating `asList()` to support them) is consistent with the `start > end` checks we're already doing, so this seems like the right fix. Looks like we also need to apply the `step < 0 ` check in `setStep()` though, in which case maybe having the constructor call `setStep()` is simpler than duplicating the checks?

Reply:
Added the extra check on `setStep()` as well.
I decided against changing the constructor to call `setStep()` as I thought it would also imply we should do the same for `setStart()` and `setEnd()`, but then you get into a bunch of details on whether we should pre-initialize the values before calling those methods, and things actually became quite difficult to read.
In the current PR, yes, you end up with duplicated checks, but at least it's fully consistent and readable.